### DISCUSSION TITLE
signin_with_twitter using API V2

### DISCUSSION THREAD

**Discussion by danx0r** [tweepy/tweepy#2020] [Questions] [answered]
Hi --

Trying to get 3rd party authentication (sign in with Twitter) to work in V2. Seems like I need to pass parameter signin_with_twitter=True in OAuth1UserHandler.get_authorization_url, but that doesn't seem possible with client.py.

in auth.py I see that if the flag is set, the endpoint switches to 'authenticate' rather than 'authorize', which apparently tells Twitter to allow a 3rd party user to use my app.

Just wondering if I have this right or I'm missing something... Note the urgency is because new dev accounts are not allowed to use api V1

Great project BTW!

**Comment by danx0r** [comment, marked as answer]
OK I appear to have something working. I use OAuth1UserHandler to do the SSO flow, then create a tweepy Client with the access codes returned by get_access_token.  Seems Essential level is allowed to do this. Might consider adding to docs - thanks

**Comment by eye0inc** [comment]
Suggestion (feature request?) add a method to client.py to get access key + secret for another user to enable signon_with_twitter

If this is possible already, let's clarify docs and add an example. Here's how I do it today using a combination of Client and direct calls to auth.py:

```
oauth1_user_handler = tweepy.OAuth1UserHandler(
    consumer_key,
    consumer_secret,
    callback="https://tweedle.social/auth"
)
auth_url = oauth1_user_handler.get_authorization_url(signin_with_twitter=True)
```
auth_url will either go directly to our app or request authentication using Twitter if not logged in already

Our callback finishes the auth1 flow (verifier_token is in the query string):

```
access_token, access_token_secret = oauth1_user_handler.get_access_token(
    verifier_token
)
```

Now we can create a Client for the authenticated user:

```
client = tweepy.Client(consumer_key=consumer_key,
                       consumer_secret=consumer_secret,
                       access_token=access_token,
                       access_token_secret=access_token_secret)

user = client.get_me()
```

**Reply by Laisky** [reply]
When accessing the `auth_url`, I encountered an issue where the webpage displays error message: 

```
Whoa there!

The request token for this page is invalid. It may have already been used, 
or expired because it is too old. Please go back to the site or application 
that sent you here and try again; it was probably just a mistake.
```

Would you happen to know what this situation is?
